src/ORM/Marshaller.php

```diff
@@ -214,6 +214,9 @@ protected function _prepareDataAndOptions($data, $options)
      */
     protected function _marshalAssociation($assoc, $value, $options)
     {
+        if (!is_array($value)) {
+            return;
```

Comment:
specially when you would rather handle this with validation and show an informative message to the user